frontend/src/features/public-form/PublicFormProvider.tsx

```diff
@@ -15,9 +21,20 @@ export const PublicFormProvider = ({
   formId,
   children,
 }: PublicFormProviderProps): JSX.Element => {
+  const [vfnTransaction, setVfnTransaction] =
+    useState<FetchNewTransactionResponse>()
   const miniHeaderRef = useRef<HTMLDivElement>(null)
-
   const { data, error, ...rest } = usePublicFormView(formId)
+  const { createTransactionMutation } = useTransactionMutations(formId)
+
+  const getTransactionId = useCallback(async () => {
+    if (!vfnTransaction || isPast(new Date(vfnTransaction.expireAt))) {
```

Comment:
let's say someone loads a form with 2 verifiable fields. they verify one immediately and leave the form open. 4 hours later, they attempt to verify the 2nd verifiable field, which causes this mutation to happen again and load a new transaction ID. however, by this time the field which was previously verified no longer has a valid signature as the transaction has expired.

since we are taking on the responsibility of refreshing the transaction, should we be dealing with this case? e.g. by automatically expiring verifications when we know the backend will reject them.

Reply:
I'll add code to handle this case
- if new transaction id is returned, reset signature map and show toast

Reply:
Resolved in later commits, please check out the implementation